py/tests/unit/with_runtime_sparkling/test_conversions.py

```diff
@@ -116,9 +118,11 @@ def testNumericRDDtoH2OFrameWithValueTooBig(spark, hc):
     min = _jvm().Long.MIN_VALUE - 1
     max = _jvm().Long.MAX_VALUE + 1
     rdd = spark.sparkContext.parallelize([1, min, max])
-    with pytest.raises(ValueError):
-        hc.asH2OFrame(rdd)
-
+    h2o_frame = hc.asH2OFrame(rdd)
+    assert h2o_frame[0, 0] == str(1)
+    assert h2o_frame[1, 0] == str(min)
```

Comment:
Are we casting unbounded Python int type to string during this conversion?

Reply:
During conversion we check the range of integeres. If it fits into int, we create Dataframe with schema containing one integer, if it fits into long, we do the same for long.

If it's bigger, we just convert the number to string and this is the case.

Reply:
Previously we failed on exception that the number is too big, but I think we can handle it like this

Reply:
WDYT about DecimalType in Spark?

Reply:
Ah, didn't know about this one. Will create JIRA for improvement for 3.30, WDYT?

Reply:
Yes, of course. I'm not pushing for that. Just mentioning as an option to think about.

Reply:
Yup, good idea